### PR TITLE
feat(business): notification toast

### PR DESCRIPTION
**Pull request by vlanz**
Notification now can be invoked through NotificationService and is placed in the bottom center of the screen by default. The vertical position can be chosen (top or bottom) whereas the horizontal position is for now blocked to the center.

### CODE REVIEW

File: src/angular-business/notification/notification-container/notification-container.component.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+@Component({
+  selector: 'sbb-notification-container',
+  templateUrl: './notification-container.component.html',
+  styleUrls: ['./notification-container.component.scss'],
+  // In Ivy embedded views will be change detected from their declaration place, rather than
+  // where they were stamped out. This means that we can't have the notification container be OnPush,
+  // because it might cause notifications that were opened from a template not to be out of date.
+  // tslint:disable-next-line:validate-decorators
+  changeDetection: ChangeDetectionStrategy.Default,
+  encapsulation: ViewEncapsulation.None,
```
Comment (kyubisation) [inline src/angular-business/notification/notification-container/notification-container.component.ts:43]: Could you try to reuse the animation from the snack-bar and see if it works?

File: src/angular-business/notification/notification/notification-config.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import { JumpMark } from './notification.component';
+
+/** Possible values for horizontalPosition on NotificationConfig. */
+export type NotificationHorizontalPosition = 'center';
+
+/** Possible values for verticalPosition on NotificationConfig. */
+export type NotificationVerticalPosition = 'top' | 'bottom';
+
+export class NotificationConfig<D = any> {
```
Comment (kyubisation) [inline src/angular-business/notification/notification/notification-config.ts:13]: Why did you remove properties when comparing with MatSnackBarConfig?
Also there is no documentation on the properties?

File: src/angular-business/notification/notification/notification.component.html [hunk 3]
```diff
@@ -1,35 +1,10 @@
...
-  <div class="sbb-notification-content-message">
-    <ng-content></ng-content>
-  </div>
-  <div *ngIf="jumpMarks">
-    <span *ngFor="let jumpMark of jumpMarks; let isLast = last" class="sbb-notification-jump-mark">
-      <a (click)="scrollTo($event, jumpMark)">{{ jumpMark.title }}</a>
-      <ng-container *ngIf="!isLast"> / </ng-container>
-    </span>
-  </div>
+<div class="sbb-notification-content-message" [innerHTML]="config.message"></div>
```
Comment (kyubisation) [inline src/angular-business/notification/notification/notification.component.html:1]: There seems to be a massive misunderstanding. The existing notification component should not be changed and remain as it was.
For the Toast Notification a new component should be created (e.g. SimpleNotification).

File: src/angular-business/notification/notification/notification-config.ts (hunk 2 above)
Comment (vlanz) [reply src/angular-business/notification/notification/notification-config.ts:13]: I opted to remove all functionalities non-specificated in the open issue / functionalities not yet provided by the "old" notification component. If you'd like a 1/1 reproduction of the SnackBar functionalities, I can still add them.

File: src/angular-business/notification/notification-container/notification-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+})
+export class NotificationContainerComponent extends BasePortalOutlet implements OnDestroy {
+  private _destroyed = false;
+  /** @docs-private */
+  @HostBinding('class.sbb-notification')
```
[the remark this answers is not in the document]
Comment (vlanz) [reply src/angular-business/notification/notification-container/notification-container.component.ts]: The lint check does fail if I do this

> Use @HostBinding or
> @HostListener rather than the `host` metadata property (https://angular.io/styleguide#style-06-03)

Or did I understand your comment wrong?

File: src/angular-business/notification/notification/notification-config.ts (hunk 2 above)
Comment (kyubisation) [reply src/angular-business/notification/notification/notification-config.ts]: I see.
Please re-add `politeness`, `announcementMessage` (replacing `message`) and `duration`. We should try to provide accessibility options where possible. (Also in `open` apply `message` to `announcementMessage`, if `announcementMessage` is empty as done in snack-bar.)
Also remove `jumpMarks`, `icon` and `readonly` for now as I don't want to provide this functionality for the toast.
Also please add documentation to the properties.

File: src/angular-business/notification/notification-container/notification-container.component.ts (hunk 1 above)
Comment (vlanz) [reply src/angular-business/notification/notification-container/notification-container.component.ts]: The opening animation now works, and as far as I can test the SnackBar as well doesn't seem to have an elaborate exit animation 👍

File: src/angular-business/notification/notification-container/notification-container.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/angular-business/notification/notification-container/notification-container.component.ts]: This has been changed on master, but is not yet done on this branch

File: src/angular-business/notification/notification-container/notification-container.component.ts (hunk 1 above)
Comment (kyubisation) [reply src/angular-business/notification/notification-container/notification-container.component.ts]: Thank you 😃 
That's fine

File: src/angular-business/notification-simple/notification-simple-container/notification-simple-container.component.scss [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+$notificationDefaultFontSize: $sizeFontDefault;
+$notificationIconMarginLeft: 16;
+$notificationIconMarginRight: 22;
+$notificationMessageMarginLeft: 8;
+$notificationMessageMarginRight: 24;
+$notificationToastOffset: 20;
+$notificationToastMaxWidth: 251;
+$notificationVerticalMargin: 12;
+
+.sbb-notification {
```
[the remark this answers is not in the document]
Comment (vlanz) [reply src/angular-business/notification-simple/notification-simple-container/notification-simple-container.component.scss]: I made the notification-container component refer to the notification.component.scss stylesheet, though I'm not sure if that's what you meant by "reuse". Since the styles are identical it does seem to make sense to use only one stylesheet though.
